addons/hr_holidays/models/hr_leave.py

```diff
@@ -630,51 +633,56 @@ def _check_double_validation_rules(self, employees, state):
-            if leave_type.validation_type == 'both':
-                self._check_double_validation_rules(employee_id, values.get('state', False))
-
-        holiday = super(HolidaysRequest, self.with_context(mail_create_nosubscribe=True)).create(values)
-        if self._context.get('import_file'):
-            holiday._onchange_leave_dates()
-        if not self._context.get('leave_fast_create'):
-            # FIXME remove these, as they should not be needed
-            if employee_id:
-                holiday.with_user(SUPERUSER_ID)._sync_employee_details()
-            if 'number_of_days' not in values and ('date_from' in values or 'date_to' in values):
-                holiday.with_user(SUPERUSER_ID)._onchange_leave_dates()
-
-            # Everything that is done here must be done using sudo because we might
-            # have different create and write rights
-            # eg : holidays_user can create a leave request with validation_type = 'manager' for someone else
-            # but they can only write on it if they are leave_manager_id
-            holiday_sudo = holiday.sudo()
-            holiday_sudo.add_follower(employee_id)
-            if holiday.validation_type == 'manager':
-                holiday_sudo.message_subscribe(partner_ids=holiday.employee_id.leave_manager_id.partner_id.ids)
-            if leave_type.validation_type == 'no_validation':
-                # Automatic validation should be done in sudo, because user might not have the rights to do it by himself
-                holiday_sudo.action_validate()
-                holiday_sudo.message_subscribe(partner_ids=[holiday_sudo._get_responsible_for_approval().partner_id.id])
-                holiday_sudo.message_post(body=_("The time off has been automatically approved"), subtype="mt_comment") # Message from OdooBot (sudo)
-            elif not self._context.get('import_file'):
-                holiday_sudo.activity_update()
-        return holiday
+            leave_types = self.env['hr.leave.type'].browse([values.get('holiday_status_id') for values in vals_list])
+            mapped_validation_type = {leave_type.id: leave_type.validation_type for leave_type in leave_types}
+
+            for values in vals_list:
+                employee_id = values.get('employee_id', False)
+                leave_type_id = values.get('holiday_status_id')
+                # Handle automatic department_id
+                if not values.get('department_id'):
+                    values.update({'department_id': self.env['hr.employee'].browse(employee_id).department_id.id})
+
+                # Handle no_validation
+                if mapped_validation_type[leave_type_id] == 'no_validation':
+                    values.update({'state': 'confirm'})
+
+                # Handle double validation
+                if mapped_validation_type[leave_type_id] == 'both':
+                    self._check_double_validation_rules(employee_id, values.get('state', False))
```

Comment:
The `default_get` logic should be considered to ensure the pre-processing of the values is correct.
```suggestion
            default_leave_type_id = self.default_get(['holiday_status_id']).get('holiday_status_id')
            leave_types = self.env['hr.leave.type'].browse([values.get('holiday_status_id') for values in vals_list])
            # NB: leave_types will contain duplicates (not sure it's worth to clean duplicates before browsing, to test)
            mapped_validation_type = {leave_type.id: leave_type.validation_type for leave_type in leave_types}
            if default_leave_type_id and default_leave_type_id not in mapped_validation_type:
                default_leave_type = self.env['hr.leave.type'].browse(default_leave_type_id)
                mapped_validation_type[default_leave_type_id] = default_leave_type.validation_type

            for values in vals_list:
                employee_id = values.get('employee_id', False)
                leave_type_id = values.get('holiday_status_id', default_leave_type_id)
                # Handle automatic department_id
                if not values.get('department_id'):
                    values.update({'department_id': self.env['hr.employee'].browse(employee_id).department_id.id})

                # Handle no_validation and double validation
                if mapped_validation_type[leave_type_id] == 'no_validation':
                    values.update({'state': 'confirm'})
                elif mapped_validation_type[leave_type_id] == 'both':
                    self._check_double_validation_rules(employee_id, values.get('state', False))
```
I only considered the defaults for `holiday_status_id`, but the same should probably be done for the `employee_id` field values.